### ISSUE TITLE
Statistic extraction from user query: StatNode

### ISSUE THREAD

**Issue by skekre98** [skekre98/NBA-Search#40]
**Description of feature/enhancement**
Function to extract statistic from user query needs to be implemented.
https://github.com/skekre98/NBA-Search/blob/738db7e4a42502ebe2d9e3617bb6f3c9a51804e2/inference/statnode.py#L50-L52
Remaining workflow for StatNode has already been implemented.

**Description of implementation**
I would recommend using `spacy` for feature extraction/recognition. There are two maps in `../data/text_data.py` labelled `total_stat_map` and `adv_stat_map`. Compare the extracted "stat" to the keys in these two maps and return the key that shows the closest similarity.  

**Why this is a good feature/enhancement**
This is necessary to return the accurate statistic for the user query. 

**Additional context**
I would recommend keeping a base threshold for similarity since the query may be misclassified. If this is the case, the function should return _None_.

Ex:

- `What is Kobe Bryant's field goal percentage?` -> `field goal percentage`
- `How many blocks does Kobe average?` -> `blocks per game`
- `What is Kobe Bryant's favorite beverage?` -> `None`

**Comment by MarTinoSVK** [comment]
Hi. I would like to work on this feature. I have some experience in Python and machine learning and I find this project really interesting. Let me know if you would allow me to contribute :)

**Comment by skekre98** [comment]
Hi @MarTinoSVK,  I will go ahead and assign the issue to you. Feel free to use this thread to discuss anything you find while working on this issue.

**Comment by MarTinoSVK** [comment]
Hey. I just wanted to discuss something. I have the feature pretty much done, but it works better with spacy pack "en_core_web_md" instead of "en_core_web_sm" since in the smaller version the similarity prediction is not very accurate and stats get often misclassified. For example stat "offensive rebounds per game" gets classified as "offensive rebound percentage" since they are similar. I would like to ask if you would consider using the medium spacy pack instead of the small one for this, or if I should work with the one already in use.

**Comment by skekre98** [comment]
I see, if you have found that the medium pack works better, please go ahead and use that one 👍.  Can you also add the medium pack to the setup script.
```
function install_spacy_cnn {
	python3 -m spacy download en_core_web_sm
        python3 -m spacy download en_core_web_md
}
```

So will you be loading the medium package into the class nlp variable?
```
class StatNode(object):

	def __init__(self):
		self.query = ""
		self.nlp = spacy.load("en_core_web_md")
```

**Comment by MarTinoSVK** [comment]
Sure thing, I will add it to the setup script.

Yes, I'll load it into the nlp variable in order to use it in the extract_stat function.

**Comment by skekre98** [comment]
Thanks! Can you also merge with upstream master. I have added some test cases for StatNode that will be run in the ci pipeline.

**Comment by skekre98** [comment]
Merged in #70
